heron/schedulers/src/java/com/twitter/heron/scheduler/local/LocalScheduler.java

```diff
@@ -261,12 +262,15 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
   @Override
   public void addContainers(Set<PackingPlan.ContainerPlan> containers) {
     synchronized (processToContainer) {
+      Set<Integer> containerIds = new HashSet<>();
+      containerIds.addAll(processToContainer.values());
       for (PackingPlan.ContainerPlan container : containers) {
-        if (processToContainer.containsKey(container.getId())) {
+        if (containerIds.contains(container.getId())) {
```

Comment:
@windie circling back on this since I realized that `HashMap.values()` returns the same view into the hashmap when called repeatedly, so the optimization question is moot. Calling remove on the map removes the entry and updates the backing values set.